main.go

```diff
@@ -185,6 +185,10 @@ func _main(ctx context.Context) error {
 		log.Error(err)
 	}
 
+	// SetPiParserValidity sets the invalid parser flag if a nil parser instance
+	// was found.
+	dcrpg.SetPiParserValidity(parser == nil)
```

Comment:
I think the solution we discussed for this with jrick would be a bit simpler:

```go
var piParser dcrpg.ProposalsFetcher
if parser != nil {
	piParser = parser
}
```

Then use `piParser` as input to `NewChainDBWithCancel`.

This avoids the need for `reflect`, and it minimizes the solution's complexity.